pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -133,13 +137,27 @@ func (inferredSpan *InferredSpan) CompleteInferredSpan(
 	})
 }
 
+// GenerateSpanId creates a secure random span id in specific scenarios, otherwise return a pseudo random id
+func GenerateSpanId() uint64 {
+	isSnapStart := os.Getenv(tags.InitType) == tags.SnapStartValue
+	if isSnapStart {
+		max := new(big.Int).SetUint64(math.MaxUint64)
+		if randId, err := rand.Int(rand.Reader, max); err != nil {
+			log.Debugf("Failed to generate a secure random span id: %v", err)
```

Comment:
how bad is it not being able to generate the spanID?
should we switch for debug to error?

Reply:
It shouldn't be too bad, since the scope of impact is isolated to inferred spans and would not cause too noticeable of a breakage if this were to fail